paparazzi/src/main/java/app/cash/paparazzi/Environment.kt

```diff
@@ -29,27 +27,20 @@ data class Environment(
   val assetsDir = "$appTestDir/src/main/assets/"
 }
 
-fun detectEnvironment(): Environment {
+fun detectEnvironment(paparazziResourcesDetailsFile: String = System.getProperty("PAPARAZZI_RESOURCES_DETAILS_FILE_KEY")): Environment {
   checkInstalledJvm()
 
   val userDir = System.getProperty("user.dir")
-  val userHome = System.getProperty("user.home")
-  val androidHome = System.getenv("ANDROID_SDK_ROOT")
-      ?: System.getenv("ANDROID_HOME")
-      ?: "$userHome/Library/Android/sdk"
-  val platformDir = Files.list(Paths.get("$androidHome/platforms"))
-      .filter { Files.isDirectory(it) }
-      .map { it.toString() }
-      .sorted()
-      .reduce { _, next -> next }
-      .orElse(null)
 
-  val configLines = File("build/intermediates/paparazzi/resources.txt").readLines()
-  val packageName = configLines[0]
-  val resDir = configLines[1]
-  val compileSdkVersion = configLines[2].toInt()
+  File(paparazziResourcesDetailsFile).readLines()
+          .run {
+            val packageName = this[0]
+            val resDir = this[1]
+            val compileSdkVersion = this[2].toInt()
+            val platformDir = this[3]
 
-  return Environment(platformDir, userDir, resDir, packageName, compileSdkVersion)
+            return Environment(platformDir, userDir, resDir, packageName, compileSdkVersion)
+          }
```

Comment:
let's remove the style changes (run{}) and trim this down to the core changes?

Reply:
I don't mind reverting back, but don't you think this is a nice Kotlin readable code?

Reply:
re: array element access vs creating of another list + popping elements, the latter doesn't really buy much.

I like using extension functions, like apply, run, etc., but this[0] doesn't read as nice IMO.

Reply:
In this case, I find it less error-prone to use `removeFirst()` rather than `list[0]`, `list[1]` etc: the chances you'll mess up the index are gone and the call order is identical to what is expected in the file.

Reply:
but, again, please let me know what you prefer here.

Reply:
If i move the vals around in modifying code, now I've inadvertently broken behavior, which would not happen with array access.

I'd prefer to leave as is.